### ISSUE TITLE
Rankfiles are broken with OMPI 

### ISSUE THREAD

**Issue by miket7777** [open-mpi/ompi#6277]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)

ompi 4.1.x  git clone (03cf3e4)

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

From git clone 

### Please describe the system on which you are running

* Operating system/version:  Red Hat Enterprise Linux Server release 7.5 (Maipo) (3.10.0-862.el7.x86_64)
* Computer hardware: Intel(R) Xeon(R) CPU E5-2697A v4 @ 2.60GHz 
* Network type: IB EDR 

-----------------------------

## Details of the problem

Please describe, in detail, the problem that you are having, including the behavior you expect to see, the actual behavior that you are seeing, steps to reproduce the problem, etc.  It is most helpful if you can attach a small program that a developer can use to reproduce your problem.

Description: rankfiles handling is broken with 4.1.x (see below). Rankfiles are working properly with earlier OMPI versions, such as, 4.0.x. 

```shell
$ cat ranks.rfile
rank 0=clx-orion-003 slot=0
rank 1=clx-orion-004 slot=1
rank 2=clx-orion-005 slot=2
rank 3=clx-orion-003 slot=18
rank 4=clx-orion-006 slot=4
rank 5=clx-orion-004 slot=25
rank 6=clx-orion-005 slot=22
rank 7=clx-orion-006 slot=27

$ mpirun -np 8 --hostfile ~/hosts/orion_4.hosts --report-bindings --rankfile ranks.rfile hostname

[clx-orion-003:13094] [[8005,0],0] ORTE_ERROR_LOG: Data unpack would read past end of buffer in file base/odls_base_default_fns.c at line 612
[clx-orion-004:06960] [[8005,0],1] ORTE_ERROR_LOG: Data unpack would read past end of buffer in file base/odls_base_default_fns.c at line 612
[clx-orion-005:31823] [[8005,0],2] ORTE_ERROR_LOG: Data unpack would read past end of buffer in file base/odls_base_default_fns.c at line 612
[clx-orion-006:21256] [[8005,0],3] ORTE_ERROR_LOG: Data unpack would read past end of buffer in file base/odls_base_default_fns.c at line 612

```

**Note**: If you include verbatim output (or a code block), please use a [GitHub Markdown](https://help.github.com/articles/creating-and-highlighting-code-blocks/) code block like below:

**Comment by jsquyres** [comment]
@miket7777 You cite "4.1" (which is master), but you cite a git hash which is on the v4.0.x branch.

Also, I believe that Mellanox is the owner of the rankfile code in Open MPI.

@yosefe @hoopoepg Is there someone there at Mellanox who can look at this?

**Comment by yosefe** [comment]
@karasevb

**Comment by rhc54** [comment]
You need to port https://github.com/open-mpi/ompi/pull/6203 over to the v4.0.x branch

**Comment by karasevb** [comment]
Closed by #6203 and #6217
